src/test/java/org/jenkinsci/plugins/p4/scm/PerforceScmSourceTest.java

```diff
@@ -0,0 +1,48 @@
+package org.jenkinsci.plugins.p4.scm;
+
+import jenkins.scm.api.SCMSource;
+import org.jenkinsci.plugins.p4.DefaultEnvironment;
+import org.jenkinsci.plugins.p4.SampleServerRule;
+import org.jenkinsci.plugins.p4.credentials.P4PasswordImpl;
+import org.jenkinsci.plugins.workflow.multibranch.WorkflowMultiBranchProject;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+
+public class PerforceScmSourceTest extends DefaultEnvironment {
+
+	private static final String P4ROOT = "tmp-ScmSourceTest-p4root";
+	private static P4PasswordImpl auth;
+
+	@ClassRule
```

Comment:
Borrow my "delete all jobs" setup method (see EventsTest in branch -api) of you are using classrule. Otherwise it is typically safer to use a regular rule so that you get a clean instance for each test